src/frontend/src/optimizer/plan_node/batch_sort.rs

```diff
@@ -32,9 +32,10 @@ pub struct BatchSort {
 
 impl BatchSort {
     pub fn new(input: PlanRef, order: Order) -> Self {
+        assert!(!order.field_order.is_empty());
         let ctx = input.ctx();
         let schema = input.schema().clone();
-        let dist = input.distribution().clone();
+        let dist = Distribution::Single;
```

Comment:
the `BatchSort` should derive input's distribution. An operator with shared distribution and a kind of order means it will order by the keys on each shard.

Reply:
and the `BatchExchange` with order property will be a merge sort exchange.

Reply:
IIUC the output of merge sort exchange is not ordered. Only ordered in each batch